### PR TITLE
Object page redesign, add object field component

### PR DESCRIPTION
**Pull request by Jibz-Mysten**
## Description 

Describe the changes or additions included in this PR.

- Added Object field view 
- Remove the old struct view and property view
- Show `ObjectFields` in the ObjectField component 
- Center loading spinner for ObjectResult page and sub-component 
- Hide the transaction component title whenever no transaction is found and update it to match the design.

## Test Plan 
preview [link](https://explorer-djhci2l4b-mysten-labs.vercel.app/object/0x2cb106b438817df75a0f7099b03c2c4e3f6d0fa7f32c366b2f91edc88dec7725) 
Screenshot 2023-04-04 at 2 06 42 PM

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/explorer/src/components/ownedobjects/views/DynamicFieldsCard.tsx [hunk 1]
```diff
@@ -152,7 +152,7 @@ export function DynamicFieldsCard({ id }: { id: string }) {
                                     {data.map((result) => (
                                         <DisclosureBox
                                             title={
-                                                <div className="max-w-[60%] truncate break-words text-body leading-relaxed font-medium text-steel-dark">
+                                                <div className="max-w-[60%] truncate break-words text-body font-medium leading-relaxed text-steel-dark">
```
Comment (Jibz-Mysten) [inline apps/explorer/src/components/ownedobjects/views/DynamicFieldsCard.tsx:155]: Setting the max width to 60% before truncating the text label text

File: apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+                                                            color="steel-darker"
+                                                        >
+                                                            {name}
+                                                        </Text>
+                                                        <div className="capitalize">
+                                                            <Text
+                                                                variant="subtitle/normal"
+                                                                color="steel"
+                                                            >
+                                                                {typeof value}
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx:185]: This is fine for now, but what we actually want here is to use `getNormalizedMoveStruct` to get the actual move types for this.

File: apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+    if (isLoading) {
+        return (
+            <div className="flex w-full justify-center">
+                <LoadingSpinner text="Loading data" />
+            </div>
+        );
+    }
+    if (isError) {
+        return null;
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx:73]: Can you display a banner component when this happens?

File: apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+                      name.toLowerCase().includes(query.toLowerCase())
+                  )
+                  .map((name) => name);
+
+    // TODO scroll to active field
+    const onChangeField = (newKeyName: string) => {
+        setActiveFieldName(newKeyName);
+    };
+
+    return fieldsNames?.length && fieldsData ? (
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx:92]: I'd probably change this to use an `if` with an early return just to keep the ternary out of the return.

File: apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+interface ObjectFieldsProps {
+    id: string;
+}
+
+interface FieldItemProps<T> {
+    value: T;
+    type?: string;
+}
+
+function FieldItem<T>({ value, type }: FieldItemProps<T>) {
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx:28]: Would _probably_ break this out into it's own file just to keep things a little cleaner

File: apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+                                                                >
+                                                                    Click to
+                                                                    view
+                                                                </Text>
+                                                            ) : (
+                                                                <FieldItem
+                                                                    value={
+                                                                        value
+                                                                    }
+                                                                    type={key}
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx:226]: Seems weird that the key is the type here? I think using the normalized module would help here

File: apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+                                <div className="max-h-[600px] min-h-full overflow-auto overflow-x-scroll py-3">
+                                    <VerticalList>
+                                        {fieldsNames.map(([name, value]) => (
+                                            <div
+                                                key={name}
+                                                className="mx-0.5 mt-0.5 md:min-w-fit"
+                                            >
+                                                <ListItem
+                                                    active={
+                                                        activeFieldName === name
```
Comment (Jordan-Mysten) [inline apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx:167]: Should probably also expand the disclosure right?

Comment (Jibz-Mysten) [reply apps/explorer/src/components/ownedobjects/views/ObjectFieldsCard.tsx:167]: Yes, I will open another PR with the feature

**Pull request closed without merging**